### ISSUE TITLE
Add the Stacked Ensemble algorithm

### ISSUE THREAD

**Issue by ledell** [h2oai/h2o4gpu#533]
We should implement stacking, as there are not many (if any) stacking implementations for GPUs.  This will require writing wrappers in Python and then adding some code to the R autogen script to port into R.  This also first requires that we enable cross-validation for all the estimators.  https://github.com/h2oai/h2o4gpu/issues/532

Scikit-learn does not yet have a stacking implementation (it seems like it may be [added](https://github.com/scikit-learn/scikit-learn/pull/8960) soon, but it's been in the works for years...).   Since we don't know when that will be complete, we could implement our own API for stacking.  The API will depend on how we enable CV of the base algorithms.

**Comment by navdeep-G** [comment]
We can always start utilizing known tools today and iterate. Thoughts?

**Comment by ledell** [comment]
@navdeep-G  Agree. I think we can write our own Stacked Ensemble API/estimator, based on how we end up enabling CV in H2O4GPU.  Later on if scikit-learn has their own SE estimator we can expose that as well if it adds additional value (e.g. multilayer stacks).
